app/src/main/java/org/schabi/newpipe/CheckForNewAppVersion.java

```diff
@@ -176,38 +171,56 @@ public static boolean isGithubApk(@NonNull final App app) {
     @Nullable
     public static Disposable checkNewVersion(@NonNull final App app) {
         final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(app);
+        final NewVersionManager manager = new NewVersionManager();
 
         // Check if user has enabled/disabled update checking
         // and if the current apk is a github one or not.
         if (!prefs.getBoolean(app.getString(R.string.update_app_key), true) || !isGithubApk(app)) {
             return null;
         }
 
+        final long expiry = prefs.getLong(app.getString(R.string.update_expiry_key), 0);
+        if (manager.isExpired(expiry)) {
+            return null;
+        }
+
```

Comment:
That does not work for me. The update check is never run on my device and returns `null` here. Why is the default value `0` and not a date in the future?

Reply:
Ah wait. Shouldn't it be 
```suggestion
        final long expiry = prefs.getLong(app.getString(R.string.update_expiry_key), 0);
        if (!manager.isExpired(expiry)) {
            return null;
        }

```